internal/status/cache.go

```diff
@@ -16,27 +16,41 @@
 package status
 
 import (
-	"time"
-
 	contour_api_v1 "github.com/projectcontour/contour/apis/projectcontour/v1"
 	"github.com/projectcontour/contour/internal/k8s"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+	gatewayapi_v1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
 )
 
-type ProxyStatus string
+// ConditionType is used to ensure we only use a limited set of possible values
+// for DetailedCondition types. It's cast back to a string before sending off to
+// HTTPProxy structs, as those use upstream types which we can't alias easily.
+type ConditionType string
 
-const (
-	ProxyStatusValid    ProxyStatus = "valid"
-	ProxyStatusInvalid  ProxyStatus = "invalid"
-	ProxyStatusOrphaned ProxyStatus = "orphaned"
-)
+// ValidCondition is the ConditionType for Valid.
+const ValidCondition ConditionType = "Valid"
+
+// AdmittedConditionValid is the valid ConditionType for Admitted.
+const AdmittedConditionValid ConditionType = "AdmittedValid"
+
+// AdmittedConditionInValid is the invalid ConditionType for Admitted.
+const AdmittedConditionInvalid ConditionType = "AdmittedInvalid"
```

Comment:
this was slightly confusing because these aren't actually condition types; they correspond to the different states ("True" or "False" for the `Admitted` condition type). Thinking about suggestions here.